assets/js/modules/tagmanager/setup.js

```diff
@@ -503,20 +566,22 @@ class TagmanagerSetup extends Component {
 							name="containers"
 							label={ __( 'Container', 'google-site-kit' ) }
 							value={ selectedContainer }
-							disabled={ hasExistingTag }
+							disabled={ hasExistingTag || [ ACCOUNT_CREATE, ACCOUNT_CHOOSE ].includes( selectedAccount ) }
 							onEnhancedChange={ this.handleContainerChange }
 							outlined
 						>
-							{ containers.concat( {
-								name: __( 'Set up a new container', 'google-site-kit' ),
-								publicId: 0,
-							} ).map( ( { name, publicId }, i ) =>
-								<Option
-									key={ i }
-									value={ publicId /* Capitalization rule exception: `publicId` is a property of an API returned value. */ }>
-									{ name }
-								</Option>
-							) }
+							{ containers
```

Comment:
Shouldn't the "Select a container" option be handled here like you do it for accounts, instead of actually adding it to the `containers` state var? I think that'd be cleaner.

Reply:
Yes, agreed.